app/code/Magento/CheckoutAgreementsGraphQl/Model/Resolver/CheckoutAgreements.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CheckoutAgreementsGraphQl\Model\Resolver;
+
+use Magento\CheckoutAgreementsGraphQl\Model\Resolver\DataProvider\CheckoutAgreements as CheckoutAgreementsDataProvider;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+
+/**
+ * CMS page field resolver, used for GraphQL request processing
+ */
+class CheckoutAgreements implements ResolverInterface
+{
+    /**
+     * @var CheckoutAgreementsDataProvider
+     */
+    private $checkoutAgreementsDataProvider;
+
+    /**
+     * @param CheckoutAgreementsDataProvider $checkoutAgreementsDataProvider
+     */
+    public function __construct(
+        CheckoutAgreementsDataProvider $checkoutAgreementsDataProvider
+    ) {
+        $this->checkoutAgreementsDataProvider = $checkoutAgreementsDataProvider;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $checkoutAgreementsData = $this->checkoutAgreementsDataProvider->getData();
```

Comment:
Need to take into account `checkout/options/enable_agreements (Enable Terms and Conditions)` (should be covered with API-functional tests)
Pay attention option value could be per `store view`